token-lending/program/src/state/reserve.rs

```diff
@@ -411,17 +411,18 @@ impl ReserveLiquidity {
     }
 
     /// Subtract borrow amount from available liquidity and add to borrows
-    pub fn borrow(&mut self, borrow_amount: Decimal, receive_amount: u64) -> ProgramResult {
-        if receive_amount > self.available_amount {
+    pub fn borrow(&mut self, borrow_decimal: Decimal) -> ProgramResult {
+        let borrow_amount = borrow_decimal.try_floor_u64()?;
```

Comment:
This should be safe because `receive_amount` is calculated the same way, and then the total borrow is effectively `receive_amount + fee_amount`.